api/authenticator.py

```diff
@@ -832,13 +832,13 @@ def authenticated_patron(self, _db, auth: Authorization):
             # The patron wants to authenticate with the
             # BasicAuthenticationProvider.
             provider = self.basic_auth_provider
-            provider_token = auth.parameters  # type: ignore[attr-defined]
+            provider_token = auth.parameters
```

Comment:
When I impremented this code the Authorization class did not have types in the werkzeug library, how is this working now?

Reply:
Good question Rishi, I should have made what I was doing here more clear. My PR description was a bit lacking 😅. These ignores are actually what prompted this PR. 

I thought `flask` included type hints, so I was wondering why we had to put in these ignores. What this lead me to is that `flask` does include type hints, but we were still installing `types-flask`, and that is overriding the included type hints, which are out of date. 

Once I removed `types-flask`, this started working, but caused some other type errors, which is what I was addressing in the rest of this PR.

Reply:
Thanks for the detailed answer! :+1: